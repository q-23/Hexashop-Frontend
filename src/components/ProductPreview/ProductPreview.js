import React from "react";

import { Link } from "react-router-dom";

import { PreviewWrapper } from "./ProductPreview.styles";
import Typography from "components/_Shared/Typography";
import Container from "components/_Shared/Container";
import FlexItem from "components/_Shared/FlexItem";
import Button from "components/_Shared/Button";
import Image from "components/_Shared/Image";

import * as styles from './ProductPreview.styles'

const ProductPreview = ({ product = {}, width = 1 }) => {
	const { name, price, image_thumbnail, _id } = product;

	return (
		<FlexItem flex_width={width} xl={3} lg={4} sm={6} xs={12} styles={styles.FLEX_ITEM_STYLES}>
			<PreviewWrapper>
				<Link to={`/product/${_id}`} >
					<Image styles={styles.IMAGE_STYLES} src={image_thumbnail && image_thumbnail.link}/>
				</Link>
				<Container styles={'height: 100%'}>
					<Link to={`/product/${_id}`} >
						<Typography styles={styles.PRODUCT_NAME_STYLES}>{name}</Typography>
					</Link>
					<Typography styles={styles.PRODUCT_PRICE_STYLES} color={'gray'}>{price}$</Typography>
					<Button with_gradient style={{bottom: 0}}>Add to cart</Button>
				</Container>
			</PreviewWrapper>
		</FlexItem>
	)
};

export default ProductPreview;