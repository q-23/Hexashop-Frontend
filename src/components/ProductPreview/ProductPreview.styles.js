import styled, { css } from 'styled-components';

export const IMAGE_STYLES = css`
	max-width: 250px; 
	width: 100%;
	height: auto;
	box-shadow: 0px 0px 41px -13px rgba(0,0,0,0.72);
	transition: .2s ease-in-out;
	cursor: pointer;
	&:hover {
		filter: saturate(100%);
		transform: scale(1.02);
	}
`;

export const FLEX_ITEM_STYLES = css`
	text-align: center;
	filter: saturate(90%);
`;

export const PRODUCT_NAME_STYLES = css`
	font-weight: 400;
	text-align: center;
	cursor: pointer;
	font-size: 1.2em;
`;

export const PRODUCT_PRICE_STYLES = css`
	text-align: center;
	font-size: .9em;
`;

export const PreviewWrapper = styled.div`
	margin: 1.5em;
`;
