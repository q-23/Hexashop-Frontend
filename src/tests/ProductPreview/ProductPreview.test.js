import React from 'react';

import ProductPreview from '../../components/ProductPreview';
import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'
import { renderWithRouter } from "../utils";

import { MOCK_PRODUCT } from "./ProductPreview.mock.data";

describe('[PRODUCT PREVIEW]', () => {
	test('Should render component', () => {
		const wrapper = render(
			renderWithRouter(<ProductPreview/>)
		);
		expect(wrapper).toMatchSnapshot();
	});

	test('Should render product name and price', () => {
		const wrapper = render(
			renderWithRouter(<ProductPreview product={MOCK_PRODUCT}/>)
		);
		expect(wrapper.getByText(MOCK_PRODUCT.name)).toBeInTheDocument();
		expect(wrapper.getByText(/23.23/)).toBeInTheDocument();
	});

	test('Should render image', () => {
		const wrapper = render(
			renderWithRouter(<ProductPreview product={MOCK_PRODUCT}/>)
		);
		expect(wrapper.getByText(MOCK_PRODUCT.name)).toBeInTheDocument();
		expect(wrapper.container.querySelector('img')).toHaveAttribute('src', MOCK_PRODUCT.image_thumbnail)
	});

	test('Should provide valid link to button', () => {
		const wrapper = render(
			renderWithRouter(<ProductPreview product={MOCK_PRODUCT}/>)
		);

		expect(wrapper.getByText(/Add to cart/)).toBeInTheDocument();
		const link = wrapper.container.querySelector(`[href="/product/${MOCK_PRODUCT._id}"]`);
		expect(link).toBeInTheDocument()
	});
})

