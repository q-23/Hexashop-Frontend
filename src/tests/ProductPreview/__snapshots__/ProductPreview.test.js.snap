// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[PRODUCT PREVIEW] Should render component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AxiKw fExOlD"
        width="md"
      >
        <div
          class="sc-AxgMl fQBKcB"
        >
          <a
            href="/product/undefined"
          >
            <img
              class="sc-AxhUy jGTwbv"
            />
          </a>
          <div
            class="sc-AxirZ kSZTkj"
          >
            <a
              href="/product/undefined"
            >
              <p
                class="sc-AxjAm lkoRYA"
              />
            </a>
            <p
              class="sc-AxjAm eLQBs"
              color="gray"
            >
              $
            </p>
            <button
              class="sc-AxhCb ijslLw"
              style="bottom: 0px;"
            >
              Add to cart
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxiKw fExOlD"
      width="md"
    >
      <div
        class="sc-AxgMl fQBKcB"
      >
        <a
          href="/product/undefined"
        >
          <img
            class="sc-AxhUy jGTwbv"
          />
        </a>
        <div
          class="sc-AxirZ kSZTkj"
        >
          <a
            href="/product/undefined"
          >
            <p
              class="sc-AxjAm lkoRYA"
            />
          </a>
          <p
            class="sc-AxjAm eLQBs"
            color="gray"
          >
            $
          </p>
          <button
            class="sc-AxhCb ijslLw"
            style="bottom: 0px;"
          >
            Add to cart
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
